spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -81,14 +84,15 @@ private CloseableIterable<ColumnarBatch> newParquetIterable(
       SparkDeleteFilter deleteFilter) {
     // get required schema if there are deletes
     Schema requiredSchema = deleteFilter != null ? deleteFilter.requiredSchema() : expectedSchema();
+    Schema vectorizationSchema = vectorizationSchema(deleteFilter);
 
     return Parquet.read(inputFile)
         .project(requiredSchema)
         .split(start, length)
         .createBatchedReaderFunc(
             fileSchema ->
                 VectorizedSparkParquetReaders.buildReader(
-                    requiredSchema, fileSchema, idToConstant, deleteFilter))
+                    vectorizationSchema, fileSchema, idToConstant, deleteFilter))
```

Comment:
Is it just `expectedSchema`?

Reply:
Not exactly. 
If no deletes, it's `expectedSchema`. 
If it's equality delete, it's `deleteFilter.requiredSchema()`, because it could be `expectedSchema` + equality filter column. For example:
```
SELECT id FROM table
```
supposed the equality delete has data == 'aaa'
then we do need to read the `data` column too, so it's `deleteFilter.requiredSchema()`, which is `id` + `data`

Reply:
In the case of pos delete, I can't use `expectedSchema` either, because it could be both pos delete and equality delete.